File: server/models/Message.go
package models

type BaseMessage struct {
	Sensitive    bool   `json:"sensitive"`
	Body         string `json:"body,omitempty"`
	AttachmentID string `json:"attachment_id,omitempty"`
}

type Message struct {
	BaseMessage
	ID             string `json:"id"`
	AttachmentName string `json:"attachment_name,omitempty"`
	AttachmentType string `json:"attachment_type,omitempty"`
	CreatedAt      int64  `json:"created_at"`
}

type InsertMessage struct {
	BaseMessage
}

var EmptyMessage = Message{}

File: web/api/models/Message.ts
interface Message {
  id: string;
  sensitive: boolean;
  body?: string;
  attachment_id?: string;
  attachment_type?: string;
  attachment_name?: string;
  created_at: number;
}

export const DefaultMessage = (): Partial<Message> => ({
  sensitive: false,
  created_at: new Date().getTime() / 1000,
});

export default Message;
